File: CS/CustomFunction_DistinctCount/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;

namespace XPG_ServerMode_EF
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            DevExpress.Data.Filtering.CriteriaOperator.RegisterCustomFunction(new DistinctCountFunction());
            InitializeComponent();
            entityServerModeSource1.QueryableSource = new NorthwindEntities().Invoices;
        }

        private void Form1_Load(object sender, EventArgs e)
        {
            pivotGridControl1.BestFit();
        }
    }

    
}

File: VB/CustomFunction_DistinctCount/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Linq
Imports System.Text
Imports System.Windows.Forms

Namespace XPG_ServerMode_EF
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			DevExpress.Data.Filtering.CriteriaOperator.RegisterCustomFunction(New DistinctCountFunction())
			InitializeComponent()
			entityServerModeSource1.QueryableSource = New NorthwindEntities().Invoices
		End Sub

		Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Load
			pivotGridControl1.BestFit()
		End Sub
	End Class


End Namespace
